panel/panel.js

```diff
@@ -1,15 +1,48 @@
+
+// Build connection to eventPage
 var eventPageConnection = chrome.runtime.connect({
   name: "panel"
-});
+}),
+  // The log for all recorded actions
+  actionLog = [];
 
+// Init or register the current panel in eventPage
 eventPageConnection.postMessage({
   name: 'init',
   tabId: chrome.devtools.inspectedWindow.tabId
 });
 
-$(function() {
-  eventPageConnection.onMessage.addListener(
-    function(request, sender, sendResponse) {
-      $('.path').text(request.path);
-    });
+eventPageConnection.onMessage.addListener(function (request, sender, sendResponse) {
+
+  var newRow = $('<tr></tr>')
+  .append('<td class="action-name">' + request.action + '</td>')
+  .append('<td class="action-path">' + request.path + '</td')
+  .append('<td class="action-mouse-pos">' + JSON.stringify(request.currentMousePos) + '</td>')
+  .append('<td class="action-key-code">' + request.keyCode + '</td>');
+
+  // Append the new request message into console
+  $('table.action-table tbody')
+  .prepend(newRow);
+
+  // Put the message into actionLog
+  actionLog.push(request);
+});
+
+// Save to file button
+$('.save-btn').click(function () {
+
+  // Create a blob file and down load it
+  // From http://bgrins.github.io/devtools-snippets/#console-save
```

Comment:
A clever way of downloading json file. From http://bgrins.github.io/devtools-snippets/#console-save